de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/followmode/FollowModeNotificationDispatcher.java

```diff
@@ -0,0 +1,66 @@
+package de.fu_berlin.inf.dpp.intellij.followmode;
+
+import de.fu_berlin.inf.dpp.editor.FollowModeManager;
+import de.fu_berlin.inf.dpp.editor.IFollowModeListener;
+import de.fu_berlin.inf.dpp.intellij.ui.Messages;
+import de.fu_berlin.inf.dpp.intellij.ui.util.NotificationPanel;
+import de.fu_berlin.inf.dpp.session.User;
+import org.apache.log4j.Logger;
+
+public class FollowModeNotificationDispatcher implements IFollowModeListener {
+
+  private static final Logger log = Logger.getLogger(FollowModeNotificationDispatcher.class);
+
+  private volatile User followedUser;
```

Comment:
IIRC all calls are done in the Swing thread so there is no need to use volatile.

Reply:
I am not certain I understand what you mean. Can you elaborate?

Reply:
Lets reverse this question. Why do you think this variable has to be volatile ?

Reply:
I thought there could be concurrency problems when the currently followed user leaves the session (which is handled by the Saros dispatcher logic) while the local user chooses to follow another user (which is handled on the IntelliJ UI thread). But you are right, as the listeners are notified  on the swing thread (or the IntelliJ equivalent) as well, this is not necessary. Will remove.